### ISSUE TITLE
Error installing diesel_cli v2.0.0-rc.0

### ISSUE THREAD

**Issue by SorenHolstHansen** [diesel-rs/diesel#3263]
### Versions

- **Rust: rustc 1.62.1 (e092d0b6b 2022-07-16)**
- **Diesel: 2.0.0-rc.0**
- **Database: Postgres**
- **Operating System: Mac with M1 chip**

### Feature Flags

- **diesel: postgres and chrono**

## Problem Description
I wanted to use the diesel-derive-enum crate with diesel 2.0.0-rc.0, and followed the issue stating that you had to download diesel_cli v2.0.0-rc.0, so i did by running
```
cargo install diesel_cli --no-default-features --features postgres --version 2.0.0-rc.0
```
but got the following output
```
Updating crates.io index
  Installing diesel_cli v2.0.0-rc.0
   Compiling proc-macro2 v1.0.43
   Compiling quote v1.0.21
   Compiling unicode-ident v1.0.3
   Compiling syn v1.0.99
   Compiling autocfg v1.1.0
   Compiling serde v1.0.142
   Compiling serde_derive v1.0.142
   Compiling version_check v0.9.4
   Compiling libc v0.2.127
   Compiling memchr v2.5.0
   Compiling tinyvec_macros v0.1.0
   Compiling pq-sys v0.4.7
   Compiling bitflags v1.3.2
   Compiling matches v0.1.9
   Compiling os_str_bytes v6.2.0
   Compiling hashbrown v0.12.3
   Compiling strsim v0.10.0
   Compiling byteorder v1.4.3
   Compiling percent-encoding v2.1.0
   Compiling unicode-bidi v0.3.8
   Compiling termcolor v1.1.3
   Compiling regex-syntax v0.6.27
   Compiling textwrap v0.15.0
   Compiling itoa v1.0.3
   Compiling ansi_term v0.12.1
   Compiling heck v0.4.0
   Compiling tinyvec v1.6.0
   Compiling proc-macro-error-attr v1.0.4
   Compiling proc-macro-error v1.0.4
   Compiling num-traits v0.2.15
   Compiling indexmap v1.9.1
   Compiling num-integer v0.1.45
   Compiling clap_lex v0.2.4
   Compiling form_urlencoded v1.0.1
   Compiling diffy v0.2.2
   Compiling unicode-normalization v0.1.21
   Compiling aho-corasick v0.7.18
   Compiling idna v0.2.3
   Compiling dirs-sys v0.3.7
   Compiling atty v0.2.14
   Compiling dirs v4.0.0
   Compiling regex v1.6.0
   Compiling dotenvy v0.15.1
   Compiling url v2.2.2
   Compiling clap v3.2.16
   Compiling chrono v0.4.20
   Compiling clap_complete v3.2.3
   Compiling diesel_derives v2.0.0-rc.1
   Compiling diesel v2.0.0-rc.1
   Compiling toml v0.5.9
   Compiling serde_regex v1.1.0
   Compiling migrations_internals v2.0.0-rc.1
   Compiling migrations_macros v2.0.0-rc.1
   Compiling diesel_migrations v2.0.0-rc.1
   Compiling diesel_cli v2.0.0-rc.0
error[E0277]: the trait bound `SelectStatement<FromClause<infer_schema_internals::information_schema::information_schema::columns::table>>: SelectDsl<(infer_schema_internals::information_schema::information_schema::columns::columns::column_name, <<Conn as diesel::Connection>::Backend as UsesInformationSchema>::TypeColumn, <<Conn as diesel::Connection>::Backend as UsesInformationSchema>::TypeSchema, infer_schema_internals::information_schema::information_schema::columns::columns::__is_nullable)>` is not satisfied
   --> /Users/sorenhansen/.cargo/registry/src/github.com-1ecc6299db9ec823/diesel_cli-2.0.0-rc.0/src/infer_schema_internals/information_schema.rs:146:1
    |
146 | / pub fn get_table_data<'a, Conn>(
147 | |     conn: &mut Conn,
148 | |     table: &'a TableName,
149 | |     column_sorting: &ColumnSorting,
...   |
209 | |             sql_types::Text,
210 | |         )> + 'static,
    | |_____________________^ the trait `SelectDsl<(infer_schema_internals::information_schema::information_schema::columns::columns::column_name, <<Conn as diesel::Connection>::Backend as UsesInformationSchema>::TypeColumn, <<Conn as diesel::Connection>::Backend as UsesInformationSchema>::TypeSchema, infer_schema_internals::information_schema::information_schema::columns::columns::__is_nullable)>` is not implemented for `SelectStatement<FromClause<infer_schema_internals::information_schema::information_schema::columns::table>>`
    |
    = note: required because of the requirements on the impl of `SelectDsl<(infer_schema_internals::information_schema::information_schema::columns::columns::column_name, <<Conn as diesel::Connection>::Backend as UsesInformationSchema>::TypeColumn, <<Conn as diesel::Connection>::Backend as UsesInformationSchema>::TypeSchema, infer_schema_internals::information_schema::information_schema::columns::columns::__is_nullable)>` for `infer_schema_internals::information_schema::information_schema::columns::table`
help: consider extending the `where` bound, but there might be an alternative better way to express this requirement
    |
210 |         )> + 'static, SelectStatement<FromClause<infer_schema_internals::information_schema::information_schema::columns::table>>: SelectDsl<(infer_schema_internals::information_schema::information_schema::columns::columns::column_name, <<Conn as diesel::Connection>::Backend as UsesInformationSchema>::TypeColumn, <<Conn as diesel::Connection>::Backend as UsesInformationSchema>::TypeSchema, infer_schema_internals::information_schema::information_schema::columns::columns::__is_nullable)>
    |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error[E0277]: the trait bound `SelectStatement<FromClause<infer_schema_internals::information_schema::information_schema::columns::table>>: SelectDsl<(infer_schema_internals::information_schema::information_schema::columns::columns::column_name, <<Conn as diesel::Connection>::Backend as UsesInformationSchema>::TypeColumn, <<Conn as diesel::Connection>::Backend as UsesInformationSchema>::TypeSchema, infer_schema_internals::information_schema::information_schema::columns::columns::__is_nullable)>` is not satisfied
   --> /Users/sorenhansen/.cargo/registry/src/github.com-1ecc6299db9ec823/diesel_cli-2.0.0-rc.0/src/infer_schema_internals/information_schema.rs:146:8
    |
146 | pub fn get_table_data<'a, Conn>(
    |        ^^^^^^^^^^^^^^ the trait `SelectDsl<(infer_schema_internals::information_schema::information_schema::columns::columns::column_name, <<Conn as diesel::Connection>::Backend as UsesInformationSchema>::TypeColumn, <<Conn as diesel::Connection>::Backend as UsesInformationSchema>::TypeSchema, infer_schema_internals::information_schema::information_schema::columns::columns::__is_nullable)>` is not implemented for `SelectStatement<FromClause<infer_schema_internals::information_schema::information_schema::columns::table>>`
    |
    = note: required because of the requirements on the impl of `SelectDsl<(infer_schema_internals::information_schema::information_schema::columns::columns::column_name, <<Conn as diesel::Connection>::Backend as UsesInformationSchema>::TypeColumn, <<Conn as diesel::Connection>::Backend as UsesInformationSchema>::TypeSchema, infer_schema_internals::information_schema::information_schema::columns::columns::__is_nullable)>` for `infer_schema_internals::information_schema::information_schema::columns::table`
help: consider extending the `where` bound, but there might be an alternative better way to express this requirement
    |
210 |         )> + 'static, SelectStatement<FromClause<infer_schema_internals::information_schema::information_schema::columns::table>>: SelectDsl<(infer_schema_internals::information_schema::information_schema::columns::columns::column_name, <<Conn as diesel::Connection>::Backend as UsesInformationSchema>::TypeColumn, <<Conn as diesel::Connection>::Backend as UsesInformationSchema>::TypeSchema, infer_schema_internals::information_schema::information_schema::columns::columns::__is_nullable)>
    |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

For more information about this error, try `rustc --explain E0277`.
error: could not compile `diesel_cli` due to 2 previous errors
error: failed to compile `diesel_cli v2.0.0-rc.0`, intermediate artifacts can be found at `/var/folders/n8/71pzt4r54g9_ff40lxj7j49w0000gn/T/cargo-installdhQULb`
```

**Comment by weiznich** [comment]
Well that's an interesting bug. To place the important thing first: Diesel_cli 2.0.0-rc.1 does not have this issue and is compatible with the rc.0 release for anything else.

Now why is this interesting: First of all [the CI passed back](https://github.com/diesel-rs/diesel/runs/6131191422?check_suite_focus=true) when we released diesel 2.0.0-rc.0. Now if I try to install diesel_cli via the provided command I can reproduce the error. The funny thing is that if I clone the repository, checkout the 2.0.0-rc.0 tag there and build diesel_cli there everything is fine. Also if I copy the source code from cargos internal cache (which is literally the code used by `cargo install`) and build that locally. I suspect an upstream issue in either rustc or cargo here due to the inconsistency in behaviour.

Closed as already fixed (2.0.0-rc.1) and/or because that's an upstream issue.

**Comment by SorenHolstHansen** [comment]
Thanks a lot! v2.0.0-rc.1 worked
